lib/extras/enc/apng.cc

```diff
@@ -195,6 +196,86 @@ void MaybeAddCICP(JxlColorEncoding c_enc, png_structp png_ptr,
   png_set_unknown_chunks(png_ptr, info_ptr, &cicp_chunk, 1);
 }
 
+bool MaybeAddSRGB(JxlColorEncoding c_enc, png_structp png_ptr,
+                  png_infop info_ptr) {
+  if (c_enc.color_space == JXL_COLOR_SPACE_RGB &&
+      c_enc.primaries == JXL_PRIMARIES_SRGB &&
+      c_enc.white_point == JXL_WHITE_POINT_D65 &&
+      c_enc.transfer_function == JXL_TRANSFER_FUNCTION_SRGB) {
+    png_set_sRGB(png_ptr, info_ptr, c_enc.rendering_intent);
+    return true;
+  }
+  return false;
+}
+
+void MaybeAddCHRM(JxlColorEncoding c_enc, png_structp png_ptr,
+                  png_infop info_ptr) {
+  if (c_enc.color_space != JXL_COLOR_SPACE_RGB) return;
+  const auto ToFixed = [](double x) {
+    return static_cast<png_fixed_point>(std::lround(PNG_FP_1 * x));
+  };
+  png_fixed_point wx, wy;
+  switch (c_enc.white_point) {
+    case JXL_WHITE_POINT_D65:
+      wx = 31270;
+      wy = 32900;
+      break;
+    case JXL_WHITE_POINT_E:
+      wx = wy = 33333;
+      break;
+    case JXL_WHITE_POINT_DCI:
+      wx = 31400;
+      wy = 35100;
+      break;
+    case JXL_WHITE_POINT_CUSTOM:
```

Comment:
I believe libjxl populates white_point_xy even if an enum value is used, although I'm not entirely sure. Double check

Reply:
Thanks, it does seem to be the case. The resulting cHRM chunks seem to have very slightly less precision, which bugs me a little, but likely not even close to an extent at which it matters.